homeassistant/components/solaredge/sensor.py

```diff
@@ -267,7 +267,8 @@ def update(self):
         """Get the latest inventory data and update state and attributes."""
         self.data_service.update()
         attr = self.data_service.attributes.get(self._json_key)
-        self._state = attr["soc"]
+        if attr and "soc" in attr:
```

Comment:
If the battery data is missing permanently we should probably avoid adding the entity at all.

Reply:
+1 but I think this is a general issue of this integration, not only for the batteries. 

Many sensors are always created independently of the inverter setup, especially power sensors:
* https://github.com/home-assistant/core/blob/dev/homeassistant/components/solaredge/sensor.py#L83
* https://github.com/home-assistant/core/blob/dev/homeassistant/components/solaredge/sensor.py#L89

Depending on the Solaredge inverter setup (with/without flow meters) these will stay `unknown` as well for some users.

Reply:
Right. So we should handle this for all sensors. :+1: